### PR TITLE
Search

### PR DESCRIPTION
**Pull request by mariuszmmm**
Added a movie and people search engine along with movie and people details.
Details:
Handling errors in case of unsuccessful server connection or calling a movie or person from outside the range of pages or calling movie or person details with a non-existing index,
Handling separate pagination for movies and people to facilitate browsing,
Implementing delayed search execution using the "debounce" function to prevent excessive requests to the server,
Handling page number or detail ID information passed as parameters to the URL,
Introducing limitations for the last page number depending on the number of found elements.

### CODE REVIEW

File: src/components/MovieTile/styled.js [hunk 1]
```diff
@@ -84,6 +84,8 @@ export const MovieTileTitle = styled.h3`
   font-size: 22px;
   line-height: 28.6px;
   text-align: left;
+  /* max-width: content;
+  overflow: hidden; */
```
Comment (Adrode) [inline src/components/MovieTile/styled.js]: Unnecessary comment, I think

File: src/components/Pagination/index.js [hunk 2]
```diff
@@ -12,46 +15,98 @@ import {
...
+  const peopleQuery = useSelector(selectPeopleQuery);
+
+  let queryPath;
+  if (moviesQuery && !peopleQuery) {
+    queryPath = `?${searchInputParamName}=${moviesQuery}`;
+  } else if (!moviesQuery && peopleQuery) {
+    queryPath = `?${searchInputParamName}=${peopleQuery}`;
+  } else {
+    queryPath = "";
+  }
```
Comment (Adrode) [inline src/components/Pagination/index.js:81]: This part is a solution that I needed, but I couldn't figure it out. Damn, great job!

File: src/components/Section/index.js [hunk 3]
```diff
@@ -3,14 +3,18 @@ import { useSelector } from "react-redux";
...
 import { useLocation } from "react-router-dom/cjs/react-router-dom.min";
 
-export const Section = ({ title, mobile, children }) => {
+export const Section = ({ title, mobile, children, noDisplay, show }) => {
   const imagesToLoad = useSelector(selectImagesToLoad);
   const location = useLocation();
-  const path = location.pathname.split("/")[1];
-
+  const path = location.pathname.split("/");
+  console.log("path", path)
```
Comment (Adrode) [inline src/components/Section/index.js]: To delete later

**Review by Adrode — APPROVED** [review]
I'm speechless. Amazing, that you've managed to work this out.

File: src/components/MovieTile/styled.js (hunk 1 above)
Comment (mariuszmmm) [reply src/components/MovieTile/styled.js]: corrected

File: src/components/Section/index.js (hunk 3 above)
Comment (mariuszmmm) [reply src/components/Section/index.js]: removed

File: src/api/fetchApi.js [hunk 4]
```diff
@@ -56,11 +55,15 @@ export const getMovieCredits = (id) => {
...
-  const updatedUrl = searchMovieUrl.replace("{query}", query).replace("{page}", page);
+  const updatedUrl = searchMovieUrl
+    .replace("{query}", query)
+    .replace("{page}", page);
   return fetchApi(updatedUrl);
 };
 
 export const getSearchPerson = (query, page) => {
-  const updatedUrl = searchMovieUrl.replace("{query}", query).replace("{page}", page);
+  const updatedUrl = searchPersonUrl
```
Comment (Urbork) [inline src/api/fetchApi.js:65]: Good you noticed that, sorry for not telling you earlier

File: src/components/Pagination/index.js (hunk 2 above)
Comment (Urbork) [reply src/components/Pagination/index.js:81]: Right, this here is impressive

File: src/features/movies/moviesSaga.js [hunk 5]
```diff
@@ -25,4 +46,6 @@ function* fetchApiHandler() {
 
 export function* moviesSaga() {
   yield takeLatest(setCurrentMoviePage.type, fetchApiHandler);
-};
\ No newline at end of file
+  yield takeLatest(setCurrentSearchPage.type, fetchApiHandler);
+  yield debounce(1000, setMoviesQuery.type, fetchApiHandler);
```
Comment (Urbork) [inline src/features/movies/moviesSaga.js:56]: I'm happy this finally works as intended

**Review by Urbork — APPROVED** [review]
Once again great piece of code. Great to have you in our team

**Comment by mariuszmmm** [comment]
To be reviewed again after changes

**Review by Urbork — APPROVED** [review]
Thanks for the changes, you are the best

**Review by Adrode — APPROVED** [review]
Looks and works great!

**Pull request merged**